apps/builder/src/features/blocks/inputs/number/components/NumberInputSettings.tsx

```diff
@@ -14,13 +37,28 @@ type Props = {
 
 export const NumberInputSettings = ({ options, onOptionsChange }: Props) => {
   const { t } = useTranslate();
+
+  useEffect(() => {
+    if (!options?.locale) {
+      const browserLocale = navigator.language;
+      if (browserLocale.match(localeRegex)) {
+        onOptionsChange({ ...options, locale: browserLocale });
+      }
+    }
+  }, []);
+
   const handlePlaceholderChange = (placeholder: string) =>
     onOptionsChange({
       ...options,
       labels: { ...options?.labels, placeholder },
     });
   const handleButtonLabelChange = (button: string) =>
     onOptionsChange({ ...options, labels: { ...options?.labels, button } });
+  const handleCurrencyChange = (currency: string) =>
+    onOptionsChange({
+      ...options,
+      currency: currency ?? defaultNumberInputOptions.currency,
```

Comment:
```suggestion
      currency: currency
```

We never save the default options, we keep it undefined and apply it at runtime

Reply:
In the case of the currency setting, leaving it undefined can crash the app

Reply:
That's why we do it at runtime:

```tsx
const numberInput = useNumberInput({
    formatOptions: {
      style: props.block.options?.style,
      currency: props.block.options?.currency ?? defaultNumberInputOptions.currency,
      unit: props.block.options?.unit,
    },
    min: minValue,
    max: maxValue,
    step: stepValue,
  });
```